homeassistant/components/recorder/util.py

```diff
@@ -543,6 +543,9 @@ def setup_connection_for_dialect(
                 or MARIA_DB_107 <= version < MARIADB_WITH_FIXED_IN_QUERIES_107
                 or MARIA_DB_108 <= version < MARIADB_WITH_FIXED_IN_QUERIES_108
             )
+
+        # Ensure all times are using UTC to avoid issues with daylight savings
+        execute_on_connection(dbapi_connection, "SET time_zone = '+00:00'")
```

Comment:
Should this be in a separate PR?

Reply:
There is a separate PR that should be merged first here

https://github.com/home-assistant/core/pull/87583